### PR TITLE
feat: support unions in arrays

### PR DESCRIPTION
**Pull request by simPod**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| Doc updated   | no
| BC breaks?    | no 
| Deprecations? | no 
| Tests pass?   | yes
| Fixed tickets | #... 
| License       | MIT

### CODE REVIEW

**Comment by scyzoryck** [comment]
@simPod I think we need also E2E test of deserialisation - I think it should already work for simple types :)

**Comment by idbentley** [comment]
My main feedback is that this PR seems rather incomplete.  

The changes to the `@Type` lexer/parser may support the simplest `array<A|B>` test case, but I can think of some edge cases that would cause that type parsing to fail.  The `@Type` typer lexer/parser currently has no union support, so this adds support for `array<A|B>` but there's no support for just `A|B`.

The only tests seem to be of the docblockparser (which is good), but no tests of any serialization/deserialization, no tests of the Type annotation lexer changes.

What is the expected behaviour of `A|B[]` in a docblock type?

Overall, I think this is a great feature, but needs some extra help.

**Comment by simPod** [comment]
This is indeed rather incomplete. I'm now stuck with DeserializationGraphNavigator that does not know how to handle unions.

File: tests/Metadata/Driver/DocBlockDriverTest.php [hunk 1]
```diff
@@ -131,6 +131,37 @@ public function testInferDocBlockMapFromGenericLikeClass()
...
+            ],
+            $m->propertyMetadata['productOrVehicleIds']->type,
+        );
+        self::assertEquals(
+            [
+                'name' => 'array',
+                'params' => [
+                    ['name' => 'int', 'params' => []],
+                    ['name' => Product::class, 'params' => []],
+                    ['name' => Vehicle::class, 'params' => []],
```
Comment (scyzoryck) [inline tests/Metadata/Driver/DocBlockDriverTest.php:150]: @simPod It seems like `array<int, string>` and `array<int|string>` will create exactly the same metadata 😓 I would consider using `Union` pseudo type same as for the unions, so it will be handled by the Union Handler :D
